### PR TITLE
[10.0][MIG] web_widget_text_markdown

### PR DESCRIPTION
**Pull request by phucngta**

### CODE REVIEW

File: web_widget_text_markdown/README.rst [hunk 1]
```diff
@@ -11,7 +11,7 @@ This module adds a new widget for text field in form view on Odoo:
 Installation
 ============
 
-It was tested on openerp trunk, 8.0 branch.
+It was tested on openerp trunk, 10.0 branch.
```
Comment (yajo) [inline web_widget_text_markdown/README.rst]: Remove this section, it makes not much sense, the addon is set to instalable, so it's implicitly expected to work

File: web_widget_text_markdown/__init__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# -*- coding: utf-8 -*-
+# © 2017 Phuc.nt - <phuc.nt@komit-consulting.com>
```
Comment (yajo) [inline web_widget_text_markdown/__init__.py]: I'd prefer to keep empty files empty

File: web_widget_text_markdown/__manifest__.py [hunk 3]
```diff
@@ -1,37 +1,25 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# © 2017 Phuc.nt - <phuc.nt@komit-consulting.com>
```
Comment (yajo) [inline web_widget_text_markdown/__manifest__.py]: Use "Copyright" word instead of symbol.

**Review by yajo — CHANGES_REQUESTED** [review]
I'd love to see demo data here, with any view implementing a markdown widget to be able to quickly test it.

**Comment by njeudy** [comment]
@Yajo I wil test this (it is a migration from my V9 module) and add some test / exemple I think

File: web_widget_text_markdown/__manifest__.py [hunk 4]
```diff
@@ -1,37 +1,28 @@
...
-##############################################################################
+# Copyright 2017 phucngta - <phuc.nt@komit-consulting.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
 {
     'name': 'web_widget_text_markdown',
-    'version': '8.0.1.0.0',
-    'author': "Sudokeys,Odoo Community Association (OCA)",
-    'maintainer': 'Sudokeys',
-    'category': '',
```
Comment (njeudy) [inline web_widget_text_markdown/__manifest__.py:26]: Do not remove other contributor (sudokeys is one of them)

File: web_widget_text_markdown/__manifest__.py [hunk 5]
```diff
@@ -1,37 +1,28 @@
...
+    'version': '10.0.1.0.0',
+    "author": "Komit, "
+              "Odoo Community Association (OCA)",
+    'category': 'Web',
     'license': 'AGPL-3',
-    'depends': ['base', 'web'],
-    'website': 'http://www.sudokey.com',
```
Comment (njeudy) [inline web_widget_text_markdown/__manifest__.py:29]: Think website should now be OCA ?

**Review by njeudy — COMMENTED** [review]
- Keep all contributors when migrating modules
- we should considere this one: https://github.com/OCA/web/pull/281
- As explain in pull 281, we should use  ace from odoo I think

**Comment by njeudy** [comment]
@yajo @phucngta : I worked on V11 to and completly remove ace (I use ace from odoo) and new js way.

See this too: https://github.com/njeudy/web_widget_text_markdown

**Review by yajo — CHANGES_REQUESTED** [review]
I took a look at the demo and saw this:

captura de pantalla de 2017-10-24 10-30-36


I have to admit the widget is pretty amazing, but please add this Less:

```less
.oe_form_field_markdown {
  width: 100%;
}
```

**Review by njeudy — APPROVED** [review]
Awesome, better than use ace editor like un V9.
Will port it to V11 very sooon.

File: web_widget_text_markdown/__manifest__.py [hunk 6]
```diff
@@ -1,37 +1,29 @@
...
-    'data': ['views/main.xml', ],
-    "qweb": ["static/src/xml/bootstrap_markdown.xml",
-             ],
-    'demo': [],
-    'installable': False,
+    'website': 'http://www.komit-consulting.com',
```
Comment (njeudy) [inline web_widget_text_markdown/__manifest__.py]: Should be https://github.com/OCA/web

**Review by njeudy — CHANGES_REQUESTED** [review]

**Comment by njeudy** [comment]
@phucngta @jcdrubay can you sign oca CLA for your devs github account, will review this quick :)

**Comment by jcdrubay** [comment]
@njeudy @OCA/core-maintainers  I think both CLA should be signed already. Is there a way to check? thanks, JC

**Review by yajo — APPROVED** [review]
The previous problem I reported is now fixed

**Review by tafaRU — APPROVED** [review]
Just tested on runbot.

**Comment by yajo** [comment]
@njeudy is your ❌ still valid?

File: web_widget_text_markdown/__manifest__.py [hunk 7]
```diff
@@ -1,37 +1,29 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright 2017 phucngta - <phuc.nt@komit-consulting.com>
```
Comment (moylop260) [inline web_widget_text_markdown/__manifest__.py]: Don't remove the original Copyright, please

File: web_widget_text_markdown/static/src/js/web_widget_text_markdown.js [hunk 8]
```diff
@@ -1,11 +1,17 @@
-openerp.web_widget_text_markdown = function (oe) {
+/* Copyright 2017 phucngta - <phuc.nt@komit-consulting.com>
```
Comment (moylop260) [inline web_widget_text_markdown/static/src/js/web_widget_text_markdown.js]: Adds other authors, please

**Review by moylop260 — CHANGES_REQUESTED** [review]

**Comment by njeudy** [comment]
@Yajo it is ok for me

**Review by njeudy — APPROVED** [review]
Ok for me code review and tests

**Comment by njeudy** [comment]
@moylop260 can you check if always red for you ?

**Comment by moylop260** [comment]
Could you use the git script from wiki: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-10.0?

**Comment by njeudy** [comment]
@moylop260 don't understand your comment .. what is wrong in this migration ?

**Comment by moylop260** [comment]
A module migrated requires all history of commits from git.
The wiki have documented how to do it

This PR have a commit so dont have a old history commits

**Comment by phucngta** [comment]
@moylop260 I tried to get old commits but it seems that does not have any commits in history. Plus, this module was already existed in branch 10.0 of OCA so i think my commit represents what i've changed in this module.

**Comment by phucngta** [comment]
When i cloned oca-web branch 10.0 to migrate module web_widget_text_markdown, there already have this module was get from 9.0 on this branch so i can't get old history commit again?
(I have updated description to FIX)
@moylop260

**Comment by njeudy** [comment]
@moylop260 @Yajo think we can merge this one as V9 was not ported so I think history is not a requierment (I was the original author of this module, and won't be hurted not to have history on this.)

I will migrate this one (with some new feature on V11) and want to keep history from now.

What do you think ?

**Review by simahawk — CHANGES_REQUESTED** [review]
ok this is not a `FIX` but a `MIG`... please, pay attention to PR titles ;)

Bare code review: LGTM.

About commit history: I don't see much important stuff to keep from the past, other contributors just add cosmetic things (like oca icons and so on).

Nevertheless I'm not sure is a good thing to break the rules for something that is really cheap to do.

**Comment by jcdrubay** [comment]
We can resubmit next week taking into account all comments. Thanks @njeudy for your flexibility, yet we need to do the right thing ;)

**Comment by yajo** [comment]
If you wish, you can squash it by author and logical changeset, but keep history please; we're not Flectra :laughing:

**Comment by njeudy** [comment]
@jcdrubay Ok ping me when you will resubmit and if you need assitance.

@Yajo : you're right .. I will wait for resubmit :) thanks all for your time.

**Comment by njeudy** [comment]
@phucngta I think you improted history in a wrong Way, I see your commit to V10.0 MIG before history .. you have to:

- Import history
- Add your MIG commit

**Comment by phucngta** [comment]
I have re-improted history @njeudy .

**Comment by njeudy** [comment]
There is a bug on runbot, I can click on group menu to test / check .. think this is not related to this migration, will investigate .

File: web_widget_text_markdown/static/src/js/web_widget_text_markdown.js [hunk 9]
```diff
@@ -82,8 +96,41 @@ openerp.web_widget_text_markdown = function (oe) {
...
+                highlight: function (code) {
+                    return hljs.highlightAuto(code).value;
+                }
+            });
+        },
+
+        _format: function(row_data, options){
+            options = options || {};
+            var markdown_text = marked(
+                oe.web.format_value(
```
Comment (njeudy) [inline web_widget_text_markdown/static/src/js/web_widget_text_markdown.js]: oe is not define think it is oe.web -> formats

File: web_widget_text_markdown/static/src/xml/bootstrap_markdown.xml [hunk 10]
```diff
@@ -1,6 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!-- vim:fdl=1:
--->
+<!-- * Copyright 2014 Sudokeys <http://www.sudokeys.com>
+ * Copyright 2017 Komit - <http:///komit-consulting.com>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).-->
```
Comment (njeudy) [inline web_widget_text_markdown/static/src/xml/bootstrap_markdown.xml]: use https://www.gnu.org/licenses/agpl (https)

File: web_widget_text_markdown/views/main.xml [hunk 11]
```diff
@@ -1,17 +1,18 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!-- vim:fdn=3:
--->
+<!-- * Copyright 2014 Sudokeys <http://www.sudokeys.com>
+ * Copyright 2017 Komit - <http:///komit-consulting.com>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).-->
```
Comment (njeudy) [inline web_widget_text_markdown/views/main.xml]: http -> https

**Review by njeudy — CHANGES_REQUESTED** [review]
Some details :)

**Review by njeudy — APPROVED** [review]
Test on runbot and code review

**Comment by njeudy** [comment]
@simahawk @moylop260 changes where done for me can you finalise your review and eventually merge ?

**Review by simahawk — APPROVED** [review]
I'd squash `[FIX] remove en.po that was erroneously created by transbot` into translation commit but LGTM

**Comment by njeudy** [comment]
@moylop260 @Yajo Is it ok for you to merge ?

**Comment by njeudy** [comment]
@Yajo can you remove availting-cla tag ? I think it's ok now
@moylop260 have you some time to finish review and merge :) thanks !

**Comment by njeudy** [comment]
@moylop260 any blocking point ?

**Comment by njeudy** [comment]
thanks @moylop260

**Pull request merged**